Untangle service from ovn requirement

booxter opened the pull request:
Note to reviewers: I advise to ignore white space changes when reviewing Functional test changes.

booxter commented on controllers/neutronapi_controller.go:
```diff
@@ -922,6 +923,14 @@ func (r *NeutronAPIReconciler) reconcileNormal(ctx context.Context, instance *ne
 	}
 
+	if instance.IsOVNEnabled() {
+		err = r.reconcileExternalOVNSecrets(ctx, helper, instance, &secretVars)
```
maybe reconcile should be executed anyway - to clean up secrets if the driver was disabled?

booxter commented on test/functional/neutronapi_controller_test.go:
```diff
@@ -1485,7 +1544,7 @@ var _ = Describe("NeutronAPI Webhook", func() {
 	})
 
-	When("A NeutronAPI instance is updated with unsupported fields", func() {
+	When("A NeutronAPI instance is configured", func() {
```
should be reverted

booxter commented on api/bases/neutron.openstack.org_neutronapis.yaml:
```diff
@@ -1938,6 +1938,11 @@ spec:
                 description: Memcached instance name.
                 type: string
+              ml2Drivers:
+                description: 'TODO: describe the option'
```
should add a proper description

booxter commented:
/ok-to-test

booxter commented:
recheck

booxter commented:
/retest-required

booxter commented:
recheck

FIPS check failure, should be ok after https://github.com/openshift/release/pull/57159

booxter commented:
/retest-required

booxter commented:
`--- PASS: kuttl/harness/neutron_no_ovn (66.29s)` 🥳

booxter commented:
@karelyatin @slawqo I think this should be ready for review now. Thanks!

slawqo commented on api/bases/neutron.openstack.org_neutronapis.yaml:
```diff
@@ -2052,6 +2052,12 @@ spec:
                 default: memcached
                 description: Memcached instance name.
                 type: string
+              ml2Drivers:
```
shouldn't we name it like e.g. "ml2MechanismDrivers" as there are different types of drivers in ML2 plugin (type drivers, mechanism drivers, extension drivers for example)

booxter commented:
/hold

booxter commented:
@karelyatin do you have any input here before I respin for what Slawek asked? (Renaming the field.)

karelyatin commented on api/v1beta1/neutronapi_types.go:
```diff
@@ -102,6 +102,10 @@ type NeutronAPISpecCore struct {
 	PreserveJobs bool `json:"preserveJobs"`
 
+	// +kubebuilder:validation:Optional
+	// Ml2Drivers - list of ml2 drivers to enable. Using ["ovn"] if not set.
```
wondering if can't we set default here itself or as part of webook instead of setting in GetMl2Drivers?

booxter commented:
@karelyatin @slawqo In the latest update, I:

- renamed the ml2Drivers field into ml2MechanismDrivers;
- implemented defaulting using CRD kubebuilder to fall back to ovn.

slawqo commented:
/lgtm for me, thx
